commit 18f253aa3f1b0693c5345756ecca7ca365c389e5
Author: Noah Watkins <nwatkins@redhat.com>
Date:   Tue Sep 4 10:17:22 2018 -0700

    mgr: monitor-controlled always on modules
    
    the original version of the always-on mgr modules feature had an issue
    such that if a user-enabled module was included in the always-on list of
    modules then after an upgrade the module would remain in the list of
    enabled modules (because it was still in the manager map). this results
    in awkward ux since the modules cannot actually be disabled. this patch
    addresses the issue by altering MgrMap to contain the always on modules,
    and centralizes control of this list in the monitor.
    
    the list of always on modules is versioned at compile time, and the
    monitor will update the manager map when this verison is bumped. manager
    daemons enable the union of user-enabled modules and the list of always
    on modules. this is important so that a user-enabled module is not
    disabled if it goes through a cycle of being in the always on list, and
    then being removed through an upgrade sequence.
    
    communicating the always on modules to the daemon enables the daemon to
    provide the best error reporting when an always on module encounters an
    issue (i.e. complain loudly).
    
    Signed-off-by: Noah Watkins <nwatkins@redhat.com>

diff --git a/src/mgr/MgrStandby.cc b/src/mgr/MgrStandby.cc
index d90b799f577..da358d868fb 100644
--- a/src/mgr/MgrStandby.cc
+++ b/src/mgr/MgrStandby.cc
@@ -188,10 +188,6 @@ void MgrStandby::send_beacon()
   // which we will transmit to the monitor.
   std::vector<MgrMap::ModuleInfo> module_info;
   for (const auto &module : modules) {
-    // do not announce always_on modules to the monitor
-    if (module->is_always_on()) {
-      continue;
-    }
     MgrMap::ModuleInfo info;
     info.name = module->get_name();
     info.error_string = module->get_error_string();
